src/content/interop/objective-c-interop.md

```diff
@@ -51,144 +52,184 @@ and then load the library with `dart:ffi`.
-The `exclude-all-by-default` flag tells `ffigen` to
-exclude everything else.
-The result is that nothing is included except `AVAudioPlayer`,
-and its dependencies, such as `NSObject` and `NSString`.
-So instead of several million lines of bindings,
-you end up with tens of thousands.
+First, we'll tell FFIgen where to find the API we're trying to
+generate bindings for.
+To do this, we set the `headers.entryPoints` option.
+
+For this example, we need to load `AVAudioPlayer.h`.
+This is part of the `AVFAudio` framework,
+which is located in your Xcode installation.
+FFIgen includes some helper functions to locate these sorts of APIs,
+such as `macSdkPath`.
+Using these helper functions makes your code generation script
+more reliable across different machines,
+which may have different SDK installation locations.
+
+`macSdkPath` returns the result of `xcrun --show-sdk-path --sdk macosx`.
+You can run this command in a terminal to locate your macOS SDKs,
+or with `--sdk iphoneos` to find your iOS SDKs.
+When generating bindings for an Apple API,
+exploring these directories is a great way to find
+the right headers to pass to FFIgen.
 
-If you need more granular control,
-you can exclude or include all declarations individually,
-rather than using `exclude-all-by-default`:
+```dart
+import 'package:ffigen/ffigen.dart';
+
+final config = FfiGenerator(
+  headers: Headers(
+    entryPoints: [
+      Uri.file(
+        '$macSdkPath/System/Library/Frameworks/AVFAudio.framework/Headers/AVAudioPlayer.h',
+      ),
+    ],
+  ),
+);
+
+void main() => config.generate();
+```
 
-```yaml
-  functions:
-    exclude:
-      - '.*'
-  structs:
-    exclude:
-      - '.*'
-  unions:
-    exclude:
-      - '.*'
-  globals:
-    exclude:
-      - '.*'
-  macros:
-    exclude:
-      - '.*'
-  enums:
-    exclude:
-      - '.*'
-  unnamed-enums:
-    exclude:
-      - '.*'
+Next, we'll define the output file.
+The main output of FFIgen is a single Dart file
+containing bindings for the given inputs.
+This file's location is defined by the `output.dartFile` option.
+
+In some cases, FFIgen will also generate a `.m` file,
+containing Objective-C code required for interop with the API.
+This file will only be generated
+if the API you're generating bindings for needs it
+(e.g. if you're using blocks or protocols).
+By default this file will have the same name as the Dart bindings,
+but with `.m` at the end of the file name.
+You can change its location with `output.objectiveCFile` option.
+If FFIgen produces this file, you must compile it into your package,
+otherwise you may get runtime exceptions to do with missing symbols.
+For this simple example we won't get a `.m` file.
+
+Another important option is `output.preamble`.
+This is simply text inserted at the top of the Dart output,
+which can be useful for adding a license header
+or disabling specific lints that FFIgen's output doesn't adhere to.
+In this case we'll disable several lints.
+
+```dart
```

Comment:
If you want to draw attention to the output section focused on here, you can highlight the relevant lines:

````suggestion
```dart highlightLines=11-16
````